tests/Unit/Parsing/OutputExceptionTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sabberworm\CSS\Tests\Unit\Parsing;
+
+use PHPUnit\Framework\TestCase;
+use Sabberworm\CSS\Parsing\OutputException;
+
+/**
+ * @covers \Sabberworm\CSS\Parsing\OutputException
+ */
+final class OutputExceptionTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function getMessageReturnsMessageProvidedToConstructor(): void
+    {
+        $message = 'The cake is a lie.';
+        $exception = new OutputException($message);
+
+        self::assertStringContainsString($message, $exception->getMessage());
+    }
+
+    /**
+     * @test
+     */
+    public function getLineNoByDefaultReturnsZero(): void
+    {
+        $exception = new OutputException('foo');
+
+        self::assertSame(0, $exception->getLineNo());
+    }
+
+    /**
+     * @test
+     */
+    public function getLineNoReturnsLineNumberProvidedToConstructor(): void
+    {
+        $lineNumber = 17;
+        $exception = new OutputException('foo', $lineNumber);
+
+        self::assertSame($lineNumber, $exception->getLineNo());
+    }
+
+    /**
+     * @test
+     */
+    public function getMessageForLineNumberProvidedIncludesMessage(): void
```

Comment:
GitHub lost my comment here.  The various 'dittos' refer to this.

I think this is conflating two tests, one that the line number is included, and one that the message is included when a line number is provided.  Also the name is confusing.  I'd suggest
- `getMessageWithLineNumberProvidedIncludesMessage` and
- `getMessageWithLineNumberProvidedIncludesLineNumber`.

Reply:
Thanks, I mixed up the tests names indeed. Fixed.

As far as the message provided to the constructor is concerned, we already have the `getMessageReturnsMessageProvidedToConstructor` tests for cases where the exception actually uses the message provided to the constructor.

Reply:
True, but we don't have a test to confirm the message when a line number argument is passed.